src/native/corehost/hostmisc/pal.unix.cpp

```diff
@@ -424,35 +417,68 @@ bool pal::get_dotnet_self_registered_dir(pal::string_t* recv)
     FILE* install_location_file = pal::file_open(install_location_file_path, "r");
     if (install_location_file == nullptr)
     {
-        trace::verbose(_X("The install_location file failed to open."));
+        trace::error(_X("The install_location file failed to open."));
         return false;
     }
 
-    bool result = false;
-
-    char buf[PATH_MAX];
-    char* install_location = fgets(buf, sizeof(buf), install_location_file);
-    if (install_location != nullptr)
+    char buffer[PATH_MAX];
```

Comment:
Given our struggles with long paths, can we not do this?
If I read the code correctly, it will effectively fail if I use a really long path and will not use it (and probably screw up the rest of the file as well).

It should be possible to write a wrapper around fgets which returns `pal::string_t` for any length.